dhis2/src/main/java/org/motechproject/dhis2/service/impl/SettingsServiceImpl.java

```diff
@@ -60,10 +62,28 @@ private synchronized void loadSettings() {
             LOGGER.debug("Loading DHIS2 settings...");
             Gson gson = new Gson();
             settings = gson.fromJson(jsonText, Settings.class);
+            if (settings.getServerURI() != null) {
+                removeTrailingSlash();
+            }
         } catch (Exception e) {
             String message = "There was an error loading json from the DHIS2 settings.";
             LOGGER.debug(message);
             throw new JsonIOException(message, e);
         }
     }
+
+    private void removeTrailingSlash() {
```

Comment:
additionally method should return parsedURI and set logic should be done above. It is more readable solution